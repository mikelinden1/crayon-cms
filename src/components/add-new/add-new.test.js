import AddNew from './add-new';
import sinon from 'sinon';

const props = {
    config: {
        itemName: 'Test'
    },
    actions: {
        openItemModal: () => {}
    }
};

it('AddNew matches snapshot', () => {
    const wrapper = shallow(<AddNew {...props} />);
    expect(wrapper).toMatchSnapshot();
});

it('Displays the correct itemName', () => {
    const propsWithJumbotron = {
        ...props,
        displayAsJumbotron: true
    };

    const wrapper = mount(<AddNew {...propsWithJumbotron} />);
    const text = wrapper.find('Button').text();

    expect(text).toEqual(`Add The First ${wrapper.props().config.itemName}`)
});

it('Calls openItemModal onClick', () => {
    const spy = sinon.spy();

    const propsWithSpy = {
        ...props,
        actions: {
            openItemModal: spy
        }
    };

    const wrapper = mount(<AddNew {...propsWithSpy} />);
    wrapper.find('button').simulate('click');

    expect(spy.calledOnce).toBeTruthy();
});