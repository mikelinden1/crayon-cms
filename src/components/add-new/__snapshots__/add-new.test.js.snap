// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNew matches snapshot 1`] = `
<Button
  className="add-new-btn btn-block"
  color="primary"
  onClick={[Function]}
  tag="button"
>
  <IconLeft
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f067",
          "M448 294.2v-76.4c0-13.3-10.7-24-24-24H286.2V56c0-13.3-10.7-24-24-24h-76.4c-13.3 0-24 10.7-24 24v137.8H24c-13.3 0-24 10.7-24 24v76.4c0 13.3 10.7 24 24 24h137.8V456c0 13.3 10.7 24 24 24h76.4c13.3 0 24-10.7 24-24V318.2H424c13.3 0 24-10.7 24-24z",
        ],
        "iconName": "plus",
        "prefix": "fas",
      }
    }
  >
    Add New
  </IconLeft>
</Button>
`;
